.github/workflows/build-and-artifact.yml

```
@@ -0,0 +1,20 @@
+jobs:
+  build-artifact:
+    runs-on: self-hosted
+
+    steps:
+      # Build the project with Maven
+      - name: Build with Maven
+        run: mvn clean install
+
+      # Upload the built artifact (e.g., .jar files) with versioning based on commit SHA
+      - name: Upload Artifacts with Versioning
+        uses: actions/upload-artifact@v3
```

actionlint (1.7.4)

the runner of "actions/upload-artifact@v3" action is too old to run on GitHub Actions. update the action's version to fix this issue

(action)